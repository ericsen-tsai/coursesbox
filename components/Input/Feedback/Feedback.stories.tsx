import { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { expect } from '@storybook/jest'
import { screen } from '@storybook/testing-library'

import Feedback from './index'

export default {
  title: 'Controls/Feedback',
  component: Feedback,
} as ComponentMeta<typeof Feedback>

export const ValidFeedback: ComponentStoryObj<typeof Feedback> = {
  play: async () => {
    await expect(screen.getByText('Looks good!')).toBeInTheDocument()
  },
  args: {
    isValid: true,
    children: 'Looks good!',
  },
}

export const InvalidFeedback: ComponentStoryObj<typeof Feedback> = {
  play: async () => {
    await expect(
      screen.getByText('Please provide a valid value')
    ).toBeInTheDocument()
  },
  args: {
    isValid: false,
    children: 'Please provide a valid value',
  },
}
