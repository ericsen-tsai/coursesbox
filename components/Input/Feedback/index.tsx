import styled from '@emotion/styled'

type Props = {
  isValid?: boolean
}

const Feedback = styled.span<Props>`
  color: ${({ isValid, theme }) =>
    isValid ? theme.font.valid : theme.font.invalid};
`

export default Feedback
