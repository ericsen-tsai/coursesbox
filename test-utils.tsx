import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
