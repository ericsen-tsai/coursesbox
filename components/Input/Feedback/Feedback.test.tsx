import { render } from '@/test-utils'

import Feedback from '.'

describe('Feedback test cases', () => {
  it('Feedback render check', () => {
    const { asFragment } = render(<Feedback isValid>Looks good!</Feedback>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Render check for invalid', () => {
    const { asFragment } = render(
      <Feedback>Please provide a valid value</Feedback>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
