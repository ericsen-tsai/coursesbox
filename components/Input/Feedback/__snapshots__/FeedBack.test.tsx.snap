// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback test cases Feedback render check 1`] = `
<DocumentFragment>
  <span
    class="css-116yo8b"
  >
    Looks good!
  </span>
</DocumentFragment>
`;

exports[`Feedback test cases Render check for invalid 1`] = `
<DocumentFragment>
  <span
    class="css-xjluvf"
  >
    Please provide a valid value
  </span>
</DocumentFragment>
`;
